### PR TITLE
NtpClock setup with Wi-Fi already initialized

### PR DESCRIPTION
**Pull request by denis-stepanov**
Contribution to address issue #24. Also provided is an example showing use of setup() without parameters.

### CODE REVIEW

File: src/ace_time/clock/NtpClock.h [hunk 1]
```diff
@@ -163,7 +163,7 @@ class NtpClock: public Clock {
     static const uint32_t kSecondsSinceNtpEpoch = 3155673600;
 
     /** Number of millis to wait during connect before timing out. */
-    static const uint16_t kConnectTimeoutMillis = 5000;
+    static const uint16_t kConnectTimeoutMillis = 10000;
```
Comment (bxparks) [inline src/ace_time/clock/NtpClock.h]: I'll change this into a configurable parameter.

File: examples/HelloNtpClock/HelloNtpClock.ino [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+static BasicZoneProcessor parisProcessor;
+NtpClock ntpClock;
+
+const char SSID[] = "...";
+const char PASSWORD[] = "...";
+
+
+void setup() {
+  delay(1000);
+  Serial.begin(115200);
```
Comment (bxparks) [inline examples/HelloNtpClock/HelloNtpClock.ino]: nit: `Serial` -> `SERIAL_PORT_MONITOR`

File: examples/HelloNtpClock/HelloNtpClock.ino [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  while (!Serial); // Wait until Serial is ready - Leonardo/Micro
```
Comment (bxparks) [inline examples/HelloNtpClock/HelloNtpClock.ino]: same here

File: examples/HelloNtpClock/HelloNtpClock.ino [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  ntpClock.setup(SSID, PASSWORD);
+  if (ntpClock.isSetup()) {
+    Serial.println("WiFi connection failed... try again.");
```
Comment (bxparks) [inline examples/HelloNtpClock/HelloNtpClock.ino]: same here

File: src/ace_time/clock/NtpClock.h [hunk 5]
```diff
@@ -68,9 +68,10 @@ class NtpClock: public Clock {
         mRequestTimeout(requestTimeout) {}
 
     /** Set up using the provided ssid and password. */
-    void setup(const char* ssid, const char* password) {
+    void setup(const char* ssid = NULL, const char* password = NULL) {
```
Comment (bxparks) [inline src/ace_time/clock/NtpClock.h]: I'll have to update the doxygen docs..

File: examples/HelloNtpClock/HelloNtpClock.ino [hunk 6]
```diff
@@ -4,7 +4,8 @@ using namespace ace_time;
 using namespace ace_time::clock;
 
 static BasicZoneProcessor parisProcessor;
-NtpClock ntpClock;
+NtpClock ntpClock, ntpClock2;
+acetime_t nowSeconds;
```
Comment (bxparks) [inline examples/HelloNtpClock/HelloNtpClock.ino]: nit: I'd move this into the `setup()` since it is used only there

Comment (denis-stepanov) [reply examples/HelloNtpClock/HelloNtpClock.ino]: Yes. It's a leftover from the version which used it in loop() as well.

**Pull request merged**